engine.py

```diff
@@ -197,6 +217,87 @@ def log_error(self, msg):
         
     
     ##########################################################################################
+    # panel interfaces
+
+    def show_panel(self, panel_id, title, bundle, widget_class, *args, **kwargs):
+        """
+        Shows a panel in Nuke. If the panel already exists, the previous panel is swapped out
+        and replaced with a new one. In this case, the contents of the panel (e.g. the toolkit app)
+        is not destroyed but carried over to the new panel.
+        
+        If this is being called from a non-pane menu in Nuke, there isn't a well established logic
+        for where the panel should be mounted. In this case, the code will look for suitable
+        areas in the UI and try to panel it there, starting by looking for the property pane and
+        trying to dock panels next to this.
+        
+        :param panel_id: Unique id to associate with the panel - normally this is a string obtained
+                         via the register_panel() call.
+        :param title: The title of the window
+        :param bundle: The app, engine or framework object that is associated with this window
+        :param widget_class: The class of the UI to be constructed. This must derive from QWidget.
+        
+        Additional parameters specified will be passed through to the widget_class constructor.
+        """
+        # note! not using the import as this confuses nuke's calback system
+        # (several of the key scene callbacks are in the main init file...)
+        import tk_nuke
+        
+        # create the panel
+        panel_widget = tk_nuke.NukePanelWidget(bundle, title, panel_id, widget_class, *args, **kwargs)
+        
+        if hasattr(tank, "_callback_from_non_pane_menu"):
```

Comment:
Could this have been solved by having a "static" on the class instead, just like you did for the UI widget earlier?